### DISCUSSION TITLE
German Voice pack

### DISCUSSION THREAD

**Discussion by truresma** [Hypfer/Valetudo#1887] [Q&A / Support] [answered]
Hello,

i want to install an German Voice pack.

in the "Misc Options" there is written "Current language: en"

I have a Xiaomi Roborock S50 on the newest firmware (Valetudo 2023.08.0.)

my wife can't speak/understand english.

i tried the following site: https://builder.dontvacuum.me/pkg/voice/rockrobo/_S5/de.pkg

and rockrobo too

copied the link from the .pkg and used the hash from here, MD5 https://emn178.github.io/online-tools/md5_checksum.html

fc7f97660d9b2f317fffcd82a24f119b

and typed in de or DE

but it is not working,

"Error installing voice pack. Check the log for details."

Log: " warn
RoborockVoicePackManagementCapability: Failed to install Voicepack. Download failed"

Thanks

**Comment by rimrul** [comment]
IIRC, I used "en" and left the hash blank when I last did it.

**Reply by truresma** [reply]
Don't work with the german one. English is working...

I have a folder with the german .wav files, but how i can make that all in a .pkg file?

https://www.file-upload.net/download-15195801/alexa_de.zip.html

or how can i access the robot via ssh?

Server: IP address of the robot
User: root
Password: empty 
Port: empty

did not work

then i can upload the .wav files manually

**Comment by kobelka** [comment]
Hello,
I have the same problem. Luckily I have two S50s and found out that it is due to the 2022.10. With the first Roborock I first installed 2022.08 and then updated directly to 2022.10. Then I tried to install the language pack de.pkg and failed as above. On the second Roborock, I installed the language pack directly after installing 2022.08 and it worked.
Is there a way to downgrade to 2022.08 to install the language pack?

**Reply by kobelka** [reply]
Maybe it depends not on the valetudo realease 2023.10, because I saw that I had the firmware 1886 on the s5 on wich  the language pack update worked...

**Comment by acidg** [comment]
You need to host the file locally, e.g. with `python -m http.server`

**Reply by audaciousengineer** [reply, marked as answer]
I second this anwer. I tried 10 ways to trick the robot running FW 2034 to load the file, to no avail.
Then, I downloaded the file manually to my computer, fired up a simple file server using
`python3 -m http.server -d /PathToFileFolder`
and pointed the robot to `http://yourIPhere:8000/file.pkg`, 
no LangCode, 
no Hash. 
Worked like a charm and would have saved me 2 hours of try-and-error if I did this from the start.

**Comment by ollie-dev** [comment]
For me with a Roborock V1 it dit work:
URL: https://builder.dontvacuum.me/pkg/voice/rockrobo/voice-v1/de.pkg
LangCode: de
Hash: 63413971f86cfd9cb18c8ed2107f3309 (Hashes are found in https://builder.dontvacuum.me/pkg/voice/rockrobo/voice-v1/pkgs.md5)

**Comment by RolandRosenfeld** [comment]
I own an S5 and run into the same issue.
I worked around it by 
1. activating an http server on the local server and placing the file there
2. using language code "va" (instead of "de"), since according to VOICEPACK_ID_TO_COUNTRY_CODE_MAPPING in backend/lib/robots/roborock/capabilities/RoborockVoicePackManagementCapability.js only the contry codes prc, tw, en and va seem to be known...

I'm not sure what brought the solution in part 1 (downgrade from https to http or using the local server), but this combo works for me.

Now the newly installed sounds are in /mnt/data/rockrobo/sounds and the bot speaks German.

**Comment by simmac** [comment]
you can find the language pack infos for the S5/S50 here: https://cnbj2.fds.api.xiaomi.com/rubys/app/voice-pkg/info
(linked here: https://robotinfo.dev/detail_roborock.vacuum.s5_0.html)

So for german, try the following:

URL: `https://cnbj2.fds.api.xiaomi.com/rubys/app/voice-pkg/package3/de.pkg`
Language: `de`
Hash (if needed): `fc7f97660d9b2f317fffcd82a24f119b`

**Reply by audaciousengineer** [reply]
I have a suspicion that this method only works for robots running FW 1886. Solution that worked for me above: https://github.com/Hypfer/Valetudo/discussions/1887#discussioncomment-8951739
